tests/Browser/Pages/Backend/Qrcodes/AdminQRCodesTest.php

```diff
@@ -0,0 +1,243 @@
+            $this->assertCount(8, $elements);
+        });
+    }
+
+    /**
+     * A Dusk test Pagination
+     *
+     * @return void
+     */
+    public function testListQRCodesWithPagination()
+    {
+        $this->makeDataOfListQRCodes(25);
+        $this->browse(function (Browser $browser) {
+            $browser->loginAs($this->adminUserToLogin)
+                    ->visit('/admin/qrcodes')
+                    ->resize(1200, 1600)
+                    ->assertSee('List QR Codes');
+            $elements = $browser->elements('.pagination li');
+            $numberPage = count($elements) - 2;
+            $this->assertTrue($numberPage == ceil(25 / (config('define.page_length'))));
+        });
+    }
+
+
+    /**
+     * Check list QR Codes with showing than 10 rows
+     *
+     * @return void
+     */
+    public function testListQRCodesHavePagination()
+    {
+        $this->makeDataOfListQRCodes(15);
+        $this->browse(function (Browser $browser) {
+            $browser->loginAs($this->adminUserToLogin)
+                    ->visit('/admin/qrcodes')
+                    ->resize(1200, 1600)
+                    ->assertSee('List QR Codes')
+                    ->click('.pagination li:nth-child(3) a');
+            $elements = $browser->elements('#table-qrcodes tbody tr');
+            $this->assertCount(5, $elements);
+            $browser->assertQueryStringHas('page', 2);
+            $this->assertNotNull($browser->element('.pagination'));
+        });
+    }
+
+    /**
+     * Test see button export QR Codes.
+     *
+     * @return void
+     */
+    public function testSeeButtonExportInListQRCodes()
+    {
+        $this->browse(function (Browser $browser) {
+            $browser->loginAs($this->adminUserToLogin)
+                    ->visit('/admin/qrcodes')
+                    ->assertSee('List QR Codes')
+                    ->assertSeeIn('.content-wrapper .content .export', 'Export')
+                    ->assertSeeLink('Export');
+        });
+    }
+
+    /**
+     * Test list QR Codes after export.
+     *
+     * @return void
+     */
+    public function testAfterExport()
+    {
+        $this->makeDataOfListQRCodes(8);
+        $this->browse(function (Browser $browser) {
+            $browser->loginAs($this->adminUserToLogin)
+                    ->visit('/admin/qrcodes')
+                    ->assertSee('List QR Codes');
+            $elementBeforeExport = $browser->elements('#table-qrcodes tbody tr');
+            $this->assertCount(8, $elementBeforeExport);
+            $browser->visit('/admin/qrcodes')
+                    ->resize(1200, 1600)
+                    ->assertSee('List QR Codes')
+                    ->clickLink('Export')
+                    ->pause(3000)
+                    ->visit('/admin/qrcodes')
+                    ->pause(3000);
+            $elementAfterExport = $browser->elements('#table-qrcodes tbody tr');
+            $this->assertCount(0, $elementAfterExport);
```

Comment:
nếu export ok thì check thêm db những qrcode đã in đã chuyển trạng thái chưa

Reply:
có kiểm tra bằng cách đếm cột trong List QR codes rồi á ( dòng 182, 183 nha )